command/agent/agent_test.go

```diff
@@ -1851,6 +1851,74 @@ func TestAgent_purgeCheckState(t *testing.T) {
 	}
 }
 
+func TestAgent_metadata(t *testing.T) {
+	config := nextConfig()
+	dir, agent := makeAgent(t, config)
+	defer os.RemoveAll(dir)
+	defer agent.Shutdown()
+
+	// Load a valid set of key/value pairs
+	config.Meta = map[string]string{
+		"key1": "value1",
+		"key2": "value2",
+	}
+	// Should succeed
+	if err := agent.loadMetadata(config); err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	agent.unloadMetadata()
+
+	// Should get error
+	config.Meta = map[string]string{
+		"": "value1",
+	}
+	if err := agent.loadMetadata(config); err == nil {
+		t.Fatalf("should have failed")
+	}
+	agent.unloadMetadata()
+
+	// Should get error
+	tooManyKeys := make(map[string]string)
+	for i := 0; i < metaMaxKeyPairs+1; i++ {
+		tooManyKeys[string(i)] = "value"
+	}
+	if err := agent.loadMetadata(config); err == nil {
+		t.Fatalf("should have failed")
+	}
+}
+
+func TestAgent_validateMetaPair(t *testing.T) {
+	longKey := fmt.Sprintf(fmt.Sprintf("%%%ds", metaKeyMaxLength+1), "")
+	longValue := fmt.Sprintf(fmt.Sprintf("%%%ds", metaValueMaxLength+1), "")
+	pairs := []struct {
+		Key     string
+		Value   string
+		Success bool
+	}{
+		// valid pair
+		{"key", "value", true},
```

Comment:
same for these - might be good to have a string that should be in the returned err (empty could mean no error)